From 1c0ba5f66e017a83a27021464475b00edb006b9c Mon Sep 17 00:00:00 2001
From: Andrew Greenberg <wizardwerdna@gmail.com>
Date: Thu, 2 Sep 2010 18:59:59 -0700
Subject: [PATCH 1/2] Modify gemspecs for activerecord, activeresoure, active support and railties
 to conform to the other gems, by adding the MIT-LICENSE to the distribution.
 Presently, while the license appears in the github repository, it is not
 distributed with the gems.

Modify the gemspec for the 'rails' gem to do likewise, and add a copy of
the MIT-LICENSE to the top-level of the repository.
---
 MIT-LICENSE |   20 ++++++++++++++++++++
 1 files changed, 20 insertions(+), 0 deletions(-)
 create mode 100644 MIT-LICENSE

diff --git a/MIT-LICENSE b/MIT-LICENSE
new file mode 100644
index 0000000..86bcb23
--- /dev/null
+++ b/MIT-LICENSE
@@ -0,0 +1,20 @@
+Copyright (c) 2004-2010 David Heinemeier Hansson
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
\ No newline at end of file
-- 
1.7.2


From 8acba10293a1f2c3934a7a171d4a743deabf9304 Mon Sep 17 00:00:00 2001
From: Andrew Greenberg <wizardwerdna@gmail.com>
Date: Thu, 2 Sep 2010 19:04:56 -0700
Subject: [PATCH 2/2] Modify gemspecs for activerecord, activesource, activesupport and railties
 so that the MIT-LICENSE will be distributed with the gems.  At present, the
 licenses exist only in the repository.  The other rails gemspecs (except for
 the rails gem itself) correctly include 'MIT-LICENSE' in the file spec.

Modify the gemspec for rails to include the MIT-LICENSE in the gem distribution
and insert a copy of the MIT-LICENSE into the root directory of the repository.
---
 activerecord/activerecord.gemspec     |    2 +-
 activeresource/activeresource.gemspec |    2 +-
 activesupport/activesupport.gemspec   |    2 +-
 rails.gemspec                         |    2 ++
 railties/railties.gemspec             |    2 +-
 5 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/activerecord/activerecord.gemspec b/activerecord/activerecord.gemspec
index 1387d6e..e0603dd 100644
--- a/activerecord/activerecord.gemspec
+++ b/activerecord/activerecord.gemspec
@@ -14,7 +14,7 @@ Gem::Specification.new do |s|
   s.homepage          = 'http://www.rubyonrails.org'
   s.rubyforge_project = 'activerecord'
 
-  s.files        = Dir['CHANGELOG', 'README.rdoc', 'examples/**/*', 'lib/**/*']
+  s.files        = Dir['CHANGELOG', 'README.rdoc', 'MIT-LICENSE', 'examples/**/*', 'lib/**/*']
   s.require_path = 'lib'
 
   s.has_rdoc         = true
diff --git a/activeresource/activeresource.gemspec b/activeresource/activeresource.gemspec
index a711687..18936cc 100644
--- a/activeresource/activeresource.gemspec
+++ b/activeresource/activeresource.gemspec
@@ -14,7 +14,7 @@ Gem::Specification.new do |s|
   s.homepage          = 'http://www.rubyonrails.org'
   s.rubyforge_project = 'activeresource'
 
-  s.files        = Dir['CHANGELOG', 'README.rdoc', 'examples/**/*', 'lib/**/*']
+  s.files        = Dir['CHANGELOG', 'README.rdoc', 'MIT-LICENSE', 'examples/**/*', 'lib/**/*']
   s.require_path = 'lib'
 
   s.has_rdoc         = true
diff --git a/activesupport/activesupport.gemspec b/activesupport/activesupport.gemspec
index df7f68f..3b533b9 100644
--- a/activesupport/activesupport.gemspec
+++ b/activesupport/activesupport.gemspec
@@ -14,7 +14,7 @@ Gem::Specification.new do |s|
   s.homepage          = 'http://www.rubyonrails.org'
   s.rubyforge_project = 'activesupport'
 
-  s.files        = Dir['CHANGELOG', 'README.rdoc', 'lib/**/*']
+  s.files        = Dir['CHANGELOG', 'README.rdoc', 'MIT-LICENSE', 'lib/**/*']
   s.require_path = 'lib'
 
   s.has_rdoc = true
diff --git a/rails.gemspec b/rails.gemspec
index 2af3016..5d6c372 100644
--- a/rails.gemspec
+++ b/rails.gemspec
@@ -15,6 +15,8 @@ Gem::Specification.new do |s|
   s.homepage          = 'http://www.rubyonrails.org'
   s.rubyforge_project = 'rails'
 
+  s.files             = Dir['MIT-LICENSE']
+
   s.bindir             = 'bin'
   s.executables        = ['rails']
   s.default_executable = 'rails'
diff --git a/railties/railties.gemspec b/railties/railties.gemspec
index adb2a76..08db9c9 100644
--- a/railties/railties.gemspec
+++ b/railties/railties.gemspec
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
   s.homepage          = 'http://www.rubyonrails.org'
   s.rubyforge_project = 'rails'
 
-  s.files              = Dir['CHANGELOG', 'README.rdoc', 'bin/**/*', 'guides/**/*', 'lib/**/{*,.[a-z]*}']
+  s.files              = Dir['CHANGELOG', 'README.rdoc', 'MIT-LICENSE', 'bin/**/*', 'guides/**/*', 'lib/**/{*,.[a-z]*}']
   s.require_path       = 'lib'
 
   s.rdoc_options << '--exclude' << '.'
-- 
1.7.2

